### PR TITLE
EN-109: fix employees reporting

### PR DESCRIPTION
**Pull request by FedericoA15**
# Description :pencil:

Restriction is added for those who can see the reporting and filtering by role

## Link to ticket :link:

https://entropyteam.atlassian.net/browse/EN-109

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested? :microscope:
image
view according to the hr manager
image
view according to admin or HR director

### CODE REVIEW

File: src/main/java/com/entropyteam/entropay/employees/repositories/EmployeeRepository.java [hunk 1]
```diff
@@ -5,11 +5,15 @@
...
 
 public interface EmployeeRepository extends BaseRepository<Employee, UUID> {
 
     List<Employee> findAllByDeletedIsFalse();
 
     List<Employee> findAllByDeletedIsFalseAndActiveIsTrue();
 
+    @Query("SELECT e FROM Employee e WHERE e.deleted = false AND e.active = true AND NOT EXISTS (SELECT r FROM e.roles r WHERE r.name = 'HR')")
+    List<Employee> findAllByDeletedIsFalseAndActiveIsTrueAndRolesNameNotLikeIgnoreCase();
+
```
Comment (agustinb4) [inline src/main/java/com/entropyteam/entropay/employees/repositories/EmployeeRepository.java:18]: try with this:
```
@Query("SELECT e FROM Employee e INNER JOIN Role r " 
            + " WHERE e.deleted = false AND e.active = true AND lower(r.name) like lower('%HR%')")
    List<Employee> findAllNotHrEmployees();
```

**Review by agustinb4 — CHANGES_REQUESTED** [review]

File: src/main/java/com/entropyteam/entropay/auth/AuthUtil.java [hunk 2]
```diff
@@ -0,0 +1,21 @@
+package com.entropyteam.entropay.auth;
+
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.core.context.SecurityContextHolder;
+
+import java.util.Collection;
+import java.util.Comparator;
+import java.util.Optional;
+
+public class AuthUtil {
```
Comment (agustinb4) [inline src/main/java/com/entropyteam/entropay/auth/AuthUtil.java:10]: It should be `AuthUtils`

**Review by agustinb4 — APPROVED** [review]
Good job!, just a minor comment

File: src/main/java/com/entropyteam/entropay/auth/AuthUtil.java (hunk 2 above)
Comment (FedericoA15) [reply src/main/java/com/entropyteam/entropay/auth/AuthUtil.java]: fixed

**Pull request merged**